src/libraries/StringParser.sol

```diff
@@ -0,0 +1,79 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import {VmSafe} from "forge-std/Vm.sol";
+import {LibString} from "@solady/utils/LibString.sol";
+
+/// @title StringParser
+/// @notice Library for parsing string representations of decimal numbers
+library StringParser {
+    using LibString for string;
+
+    /// @notice Struct to hold parsed amount components
+    struct AmountComponents {
+        uint256 wholePart;
```

Comment:
I personally prefer the following: 
- Integer part (instead of wholePart)
- Fractional part (instead of decimalPart)

I think it's more common to head the above than what is in this PR. But I could be wrong. I had to look up `wholePart` meaning.